### PR TITLE
Switched to reactive forms, added min/max validations

### PR DESCRIPTION
**Pull request by Shpota**

### CODE REVIEW

File: webapp/src/app/student-dialog/student-dialog.component.ts [hunk 1]
```diff
@@ -38,6 +47,14 @@ export class StudentDialog implements OnDestroy {
       .subscribe(this.dialogRef.close);
   }
 
+  hasError(controlName: string, errorCode: string): boolean {
+    return !this.controlGroup.valid && this.controlGroup.hasError(errorCode, [controlName]);
+  }
+
+  formValue(controlName: string) {
```
Comment (ForEasyWork) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: This method  can be private

Comment (ForEasyWork) [reply webapp/src/app/student-dialog/student-dialog.component.ts]: Add returned type to method signature

File: webapp/src/app/student-dialog/student-dialog.component.ts [hunk 2]
```diff
@@ -21,9 +21,18 @@ export class StudentDialog implements OnDestroy {
...
+  ) {
+    this.controlGroup = new FormGroup({
+      name: new FormControl(student.name, Validators.required),
+      age: new FormControl(student.age, [
+        Validators.required, Validators.min(0), Validators.max(100)
+      ])
+    });
+  }
 
   save() {
```
Comment (ForEasyWork) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: `save()` => `save(): void`

File: webapp/src/app/student-dialog/student-dialog.component.ts [hunk 3]
```diff
@@ -21,9 +21,18 @@ export class StudentDialog implements OnDestroy {
     @Inject(MAT_DIALOG_DATA) public student: Student,
     public dialogRef: MatDialogRef<Student>,
```
Comment (ForEasyWork) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: `dialogRef: MatDialogRef<Student>` => `dialogRef: MatDialogRef<StudentDialog>`

File: webapp/src/app/student-dialog/student-dialog.component.ts (hunk 1 above)
Comment (shpotainna) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: This method can be private

File: webapp/src/app/student-dialog/student-dialog.component.ts (hunk 2 above)
Comment (shpotainna) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: `save()` => `save(): void`

File: webapp/src/app/student-dialog/student-dialog.component.ts (hunk 3 above)
Comment (shpotainna) [inline webapp/src/app/student-dialog/student-dialog.component.ts]: `dialogRef: MatDialogRef<Student>` => `dialogRef: MatDialogRef<StudentDialog>`

**Pull request merged**